### PR TITLE
Profile for Electron Cash

### PR DESCRIPTION
**Pull request by haplo**
This is a profile for the [Electron Cash](https://electroncash.org/) Bitcoin Cash SPV client.

### CODE REVIEW

File: etc/profile-a-l/electron-cash.profile [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Firejail profile for electron-cash
+# Description: Lightweight Bitcoin Cash wallet
+# This file is overwritten after every install/update
+# Persistent local customizations
+include electron-cash.local
+# Persistent global definitions
+include globals.local
+
+noblacklist ${HOME}/.electron-cash
```
Comment (glitsj16) [inline etc/profile-a-l/electron-cash.profile:9]: Firejail's logic assumes a `noblacklist foo` is warranted when a `blacklist foo` exists somewhere in our other included files. So you'll need to add `blacklist ${HOME}/.electron-cash` to `disable-programs.inc` to protect that path from being accessible in other application sandboxes.

File: etc/profile-a-l/electron-cash.profile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+novideo
+protocol unix,inet,inet6
+seccomp
+
+disable-mnt
+private-bin electron-cash,python*
+private-cache
+?HAS_APPIMAGE: ignore private-dev
+private-dev
+private-etc alternatives,ca-certificates,crypto-policies,dconf,fonts,ld.so.cache,ld.so.preload,machine-id,pki,resolv.conf,ssl
```
Comment (glitsj16) [inline etc/profile-a-l/electron-cash.profile:48]: There's a new private-etc format now: https://github.com/netblue30/firejail/discussions/5610. Please use that for new profiles. In this case the line should look like:
`private-etc @network,@tls-ca,@x11`.

**Review by glitsj16 — CHANGES_REQUESTED** [review]
Thanks for the profile! Some smallish changes are needed, check the comments please.

File: etc/profile-a-l/electron-cash.profile (hunk 1 above)
Comment (haplo) [reply etc/profile-a-l/electron-cash.profile:9]: That makes sense, I added the `blacklist` to `disable-programs.inc`.

File: etc/profile-a-l/electron-cash.profile (hunk 2 above)
Comment (haplo) [reply etc/profile-a-l/electron-cash.profile:48]: I tested this change and electron-cash fails to find `/etc/ssl/certs/ca-certificates.crt`.

Do I need master version of firejail?

```
$ firejail --version
firejail version 0.9.72

Compile time support:
- always force nonewprivs support is disabled
- AppArmor support is enabled
- AppImage support is enabled
- chroot support is enabled
- D-BUS proxy support is enabled
- file transfer support is enabled
- firetunnel support is disabled
- IDS support is disabled
- networking support is enabled
- output logging is enabled
- overlayfs support is disabled
- private-home support is enabled
- private-cache and tmpfs as user enabled
- SELinux support is disabled
- user namespace support is enabled
- X11 sandboxing support is enabled
```

Comment (glitsj16) [reply etc/profile-a-l/electron-cash.profile:48]: > Do I need master version of firejail?

Correct. This private-etc refactoring came in after 0.9.72. It's always good to use the git build when developing new profiles. Not strictly required, but saves some time and hair-pulling for code refactorings.

**Review by glitsj16 — APPROVED** [review]
Thanks for the changes!

File: etc/profile-a-l/electron-cash.profile (hunk 2 above)
Comment (haplo) [reply etc/profile-a-l/electron-cash.profile:48]: `firejail-git` is in Arch Linux's AUR so it'd be easy for me to install, I just don't want it to mess with my existing system. Is there any guide for developers?

I'm thinking setting up a VM might be best, although that would probably hide some hardware-access related issues. A Vagrantfile that sets everything up for development would be great IMO.

Comment (glitsj16) [reply etc/profile-a-l/electron-cash.profile:48]: > Is there any guide for developers?

There are a few entrees on the [wiki](https://github.com/netblue30/firejail/wiki). Especially the ones under `firejail for contributors`.

Vagrant support is always welcome. I haven't seen anything from the regular contributors on that though. You could open a feature request for it.

**Pull request merged**